pkg/crc/validation/validation.go

```diff
@@ -98,6 +110,35 @@ func ValidateIPAddress(ipAddress string) error {
 	return nil
 }
 
+func ValidateURL(inputString string) error {
+	u, err := url.ParseRequestURI(inputString)
+	if err != nil {
+		logging.Debugf("Failed to parse url: %v", err)
+		return err
+	}
+	switch {
+	case u.Scheme == "http", u.Scheme == "https":
+		return nil
+	case u.Scheme == "docker":
+		if len(strings.Split(inputString, ":")) != 3 {
+			return fmt.Errorf("invalid %s registry URL, tag is required (such as docker://quay.io/crcont/ocp-bundle:4.11.0)", inputString)
+		}
+		return nil
+	default:
+		return &url.Error{Op: "parse", URL: inputString, Err: fmt.Errorf("invalid %s format (only supported http, https or docker)", inputString)}
+	}
+}
+
+func validBundleForPreset(inputString string, preset crcpreset.Preset) error {
+	if strings.HasPrefix(inputString, "docker://") || preset == crcpreset.OpenShift {
+		return nil
+	}
+	if !strings.Contains(filepath.Base(inputString), preset.String()) {
+		return fmt.Errorf("invalid %s bundle for %s preset", filepath.Base(inputString), preset)
+	}
```

Comment:
> I'd prefer not to rush this and to take time to do it properly. I don't know why with an OpenShift preset it's ok to use `crc_okd_....crcbundle` or `crc_podman_....crcbundle`, 

This is handled, now we are checking the bundle prefix not just the preset string so that even hypervisor side also taken care.

> I don't know why `https://internalmirror/podman-dev-env/crc_okd....crcbundle` is acceptable, ...

I don't think we should force our user to tell what `org` they want to use what if the use existing org to push bundle there. What we can do is to just make sure the `image` name should be something we have `okd-bundle`, `openshift-bundle` ..etc.

Reply:
> This is handled, now we are checking the bundle prefix not just the preset string so that even hypervisor side also taken care.

It's still quite inconsistent between  docker:// and the other schemes, which I don't like. For https/file, we trust the URL to get the bundle name, for docker://, we do something complicated, we infer the bundle name from the preset, totally ignoring the docker image name, but we still trust the version from it. I'd prefer if in this case to we also trusted the URL and went from `podman-bundle:4.1.1` to `crc_podman_libvirt_4.1.1.crcbundle` rather than `crc_$currentPreset_libvirt_4.1.1.crcbundle` I've tried implementing this in https://github.com/cfergeau/crc/commits/issue_3277

Then we can add a function checking if the name we built from the URL matches the name we expect given the current preset, and it will now be done in a similar way for docker/http/file. I've dropped it for now as imo it can be a followup.